### DISCUSSION TITLE
KineticsToolkit extracts angle value error from c3d data

### DISCUSSION THREAD

**Discussion by gaozelle** [kineticstoolkit/kineticstoolkit#146] [1. Users Q&A] [answered]
After using vicon to collect kinematics data, I found that all kinds of angle data in the c3d data extracted with kineticsToolkit were incorrect. But the EMG and point data are correct.

- The output of the kineticsToolkit looks like this:
```
zzt_moka["Points"].plot("RKneeAngles")
```
![image](https://user-images.githubusercontent.com/31237218/222888228-052d198e-84d4-4e24-8f4c-b4a67cef8eca.png)

- I used pyomeca to extract the angle data of c3d, which is correct:
```
import pyomeca
from pyomeca import Markers
tdata_path = "/Users/macgazelle/Desktop/c3d/zzt-mokka.c3d"
tmarkers = Markers.from_c3d(tdata_path, prefix_delimiter=":")
tmarkers.sel(channel="RKneeAngles", axis = "z").plot.line(x="time")
```
![image](https://user-images.githubusercontent.com/31237218/222886718-c5c6e6dc-fe19-4827-a73d-172e8c4ad625.png)

- It is also correct to use mokka to read the angle of c3d data:
![6751677920446_ pic](https://user-images.githubusercontent.com/31237218/222886889-9ddd7730-5b44-4b1f-bd47-e97ccd0c7557.jpg)

I don't know what the problem is. Please tell me how to convert to the correct angle value. If you have time, please upgrade the kineticsToolkit to calculate the correct angle value.

I'd be happy to help if I could.
Thank you very much.
Thank you also for the kineticsToolkit.

**Comment by felixchenier** [comment]
@gaozelle Thank you for noticing and testing with the other softwares. Would you mind sending me this c3d, either by email (chenier dot felix at uqam dot ca) or right here in this thread? It will greatly help me. Tx

**Comment by felixchenier** [comment, marked as answer]
Hi @gaozelle 

Thanks for the question and c3d file. This may indeed be described as a bug in Kinetics Toolkit and I will open an issue related to this. The workaround is simple though (see below).

The problem is that `ktk.read_c3d` doesn't distinguish points between proper points (markers) and computed "points" such as angles and forces. Since the point unit in your c3d is millimeters and the default in KTK is to convert to meters, then it divides every point coordinate by 1000, including the computed angles, forces, moments, etc.

While I figure out a clean way to address this issue, you can either multiply the angles by 1000, or load the file "as is" without any conversion using `ktk.read_c3d(filename, convert_point_unit=False)`.
